### PR TITLE
Cdev 383 expand interaction tracking to all components

### PR DESCRIPTION
**Pull request by melocule**
CDEV-383

Adds typing and more consistency in the tracking names. Also tracks some more stuff.

### CODE REVIEW

File: src/SecuredApp.tsx [hunk 1]
```diff
@@ -24,7 +26,10 @@ const SecuredAppComponent = ({ AppComponent }: SecuredAppProps) => {
 
       <button
         type="button"
-        onClick={() => logout({ returnTo })}
+        onClick={() => {
+          trackInteraction("logout");
+          logout({ returnTo });
+        }}
```
Comment (mrosata) [inline src/SecuredApp.tsx:32]: One thing to check for here is that "logout" events actually make it into Amplitude. This concern may turn out not to be valid, but if the Auth0 logout redirect happens immediately before the browser makes the analytic POST request to ctw with the "logout" event then we might have to change this to await the sending of the analytic event before calling `logout({ returnTo })`.

If you do need to do this, there's a helper `const { analyticsEvent } = useAnalytics` that calls the analytics service directly and can be awaited.

https://github.com/zeus-health/ctw-component-library/blob/6349b423df8a3f9535719817500564e599233a94/src/components/core/providers/analytics/use-analytics.ts#L27-L29
> side note: the helper method `analyticsEvent` doesn't put the componentName and upid on the event, which is probably okay in this situation anyways.

File: src/components/content/CCDA/ccda_viewer/components/Section/Section.tsx [hunk 2]
```diff
@@ -5,16 +5,21 @@ import { useState } from "react";
...
   const [isOpen, setIsOpen] = useState(true);
+  const { trackInteraction } = useAnalytics();
 
   if (!humanReadable) return null;
 
-  const handleClick = () => setIsOpen(!isOpen);
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+    trackInteraction(isOpen ? "collapse_document_section" : "expand_document_section");
+  };
```
Comment (mrosata) [inline src/components/content/CCDA/ccda_viewer/components/Section/Section.tsx:22]: We could make the event of clicking the document section a single event where "collapse" and "expand" are metadata on the event.
  
```suggestion
  const handleClick = () => {
    setIsOpen(!isOpen);
    trackInteraction("toggle_document_section", {
      direction: isOpen ? "collapse" : "expand",
    });
  };
```

File: src/components/content/observations/helpers/trends.tsx [hunk 3]
```diff
@@ -53,13 +53,13 @@ export const ObservationTrends = ({ model }: ObservationTrendsProps) => {
                       })}
                       onClick={() => {
                         if (trend.diagnosticReport) {
-                          trackInteraction("open_diagnostic_report");
+                          trackInteraction("open_diagnostic_report", { target: "btn" });
```
Comment (mrosata) [inline src/components/content/observations/helpers/trends.tsx:56]: nice idea 👍🏼

File: src/components/content/resource/helpers/history-entry.tsx [hunk 4]
```diff
@@ -84,9 +91,7 @@ const DetailSummary = ({
       aria-label="details"
       onClick={() => {
         setIsDetailShown(!isDetailShown);
-        trackInteraction("expand_detail", {
-          direction: isDetailShown ? "collapse" : "expand",
-        });
+        trackInteraction(isDetailShown ? "collapse_history_entry" : "expand_history_entry");
```
Comment (mrosata) [inline src/components/content/resource/helpers/history-entry.tsx:94]: This is same as a previous comment. Is there a reason we wouldn't want to group these?
```suggestion
        trackInteraction("toggle_detail", {
          direction: isDetailShown ? "collapse" : "expand",
        });
```

File: src/components/core/providers/analytics/use-analytics.ts [hunk 5]
```diff
@@ -2,6 +2,65 @@ import { useContext, useMemo } from "react";
 import { AnalyticsContext } from "./context";
 import { usePatient } from "@/components/core/providers/patient-provider";
 import { useTelemetry } from "@/components/core/providers/telemetry/use-telemetry";
+import { SortDir } from "@/utils/sort";
+
+// Tracked events indicate the effect the user initiated.
+// They don't indicate details like patient, overarching component, or cause, which belong in metadata.
+type Action =
```
Comment (mrosata) [inline src/components/core/providers/analytics/use-analytics.ts:9]: good idea

File: src/SecuredApp.tsx (hunk 1 above)
Comment (melocule) [reply src/SecuredApp.tsx:32]: Just realized this button isn't part of the CTW app

**Review by mrosata — DISMISSED** [review]

**Review by mrosata — APPROVED** [review]

**Pull request merged**